### PR TITLE
Add function to reliably acquire message body length

### PR DESCRIPTION
**Pull request by lizhanhui**
## What is the purpose of the change

Add function to reliably acquire message body length

Fix a few typo and grammar error

## Brief changelog

CMessage.{h,cpp}, CMessageExt.{h,cpp}

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before requesting the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when a cross-module dependency exists.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

File: test/src/extern/CMessageExtTest.cpp [hunk 1]
```diff
@@ -41,8 +41,11 @@ TEST(cmessageExt, info) {
   mqMessageExt->setKeys("testKeys");
   EXPECT_EQ(GetMessageKeys(messageExt), mqMessageExt->getKeys());
 
-  mqMessageExt->setBody("testBody");
+  std::string body("testBody");
+  body.append(3, '\0');
+  mqMessageExt->setBody(body.c_str(), body.length());
   EXPECT_EQ(GetMessageBody(messageExt), mqMessageExt->getBody());
```
Comment (ifplusor) [inline test/src/extern/CMessageExtTest.cpp:47]: This case is ok?

File: test/src/extern/CMessageTest.cpp [hunk 2]
```diff
@@ -14,6 +14,8 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+#include <cstring>
+#include <string>
 #include "gmock/gmock.h"
```
Comment (ifplusor) [inline test/src/extern/CMessageTest.cpp:19]: Need a LF.

File: test/src/extern/CMessageExtTest.cpp [hunk 3]
```diff
@@ -14,7 +14,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
+#include <string>
 #include "gmock/gmock.h"
```
Comment (ifplusor) [inline test/src/extern/CMessageExtTest.cpp:18]: Need a LF.

File: .gitignore [hunk 4]
```diff
@@ -6,3 +6,5 @@ libs/signature/lib
 tmp_*
 Testing
 .vscode
+.cache
```
Comment (ifplusor) [inline .gitignore:9]: .cache/ ?

**Review by ifplusor — CHANGES_REQUESTED** [review]

File: test/src/extern/CMessageExtTest.cpp (hunk 1 above)
Comment (lizhanhui) [reply test/src/extern/CMessageExtTest.cpp:47]: The message body needs to support arbitrary binary data.

File: .gitignore (hunk 4 above)
Comment (lizhanhui) [reply .gitignore:9]: Trailing '/' is optional.

File: test/src/extern/CMessageTest.cpp (hunk 2 above)
Comment (lizhanhui) [reply test/src/extern/CMessageTest.cpp:19]: OK

File: test/src/extern/CMessageExtTest.cpp (hunk 3 above)
Comment (lizhanhui) [reply test/src/extern/CMessageExtTest.cpp:18]: OK

**Pull request merged**